ql/patterns/observable.hpp

```diff
@@ -410,17 +412,23 @@ namespace QuantLib {
       private:
         ObservableSettings() : updatesType_(UpdatesEnabled) {}
 
+#if defined(QL_USE_STD_SHARED_PTR)
+        typedef std::set<ext::weak_ptr<Observer::Proxy>,
```

Comment:
Maybe we can put `owner_less` in the `ext` namespace too?

Reply:
I was actually looking into this but the drawback is that nearly all code will have to include 
#include <boost/smart_ptr/owner_less.hpp>
and this would be against Peter's efforts to reduce compile time.